docs/odcs.rst

```diff
@@ -24,84 +24,214 @@ idea is to not require the user to provide a yum repo url when building
 container images.  Instead, OSBS will use ODCS to generate a yum repo with the
 relevant signed packages based on user configuration.
 
+Concepts
+--------
+
+Signing Intent
+""""""""""""""
+
+When an ODCS compose is used, the idea of a signing intent is used. This refers
+to the signature of the content to be included in the container image. For
+instance, an environment may provide the following intents: ``release``,
+``beta``, and ``unsigned``. Each one of those intents is then mapped to a list
+of signing keys. These signing keys are then used during compose creation. The
+packages to be included must have been signed by any of the keys listed. In the
+example above, the intents could be mapped to the following keys::
+
+    # Only include packages that have been signed by "my-release-key"
+    release -> my-release-key
+    # Include packages that have been signed by either "my-beta-key" or
+    # "my-release-key"
+    beta -> my-beta-key, my-release-key
+    # Do not check signature of packages - may include unsigned packages
+    unsigned -> <empty>
+
+This mapping will be defined in the secret named by reactor_config_secret which
+will allow updates to be reflected instantly for automatically triggered builds
+from an existing BuildConfig. For example::
+
+    signing_intents:
+    - release:
+      - 'my-release-key'
+    - beta:
+      - 'my-beta-key'
+      - 'my-release-key'
+    - unsigned: []
+
+The definition of intents also describes the restrictive order, which will be
+enforced when building layered images. For instance, consider the case of two
+images, X and Y. Y uses X as its parent image (FROM X). If image X was built
+with "beta" intent, image Y's intent can only be "beta" or "unsigned". If the
+dist-git repo for image Y has it configured to use "release" intent, this value
+will be downgraded to "beta" at build time.
+
+Automatically downgrading the signing intent, instead of failing the build, is
+important for allowing a hierarchy of layered images to be built automatically
+by ImageChangeTriggers. For instance, with Continuous Integration in mind, a
+user may want to perform daily builds without necessarily requiring signed
+packages while periodically also producing builds with signed content. In this
+case, the ``signing_intent`` in ``container.yaml`` can be set to ``release`` for
+all the images in hierarchy.  Whether or not the layered images in the hierarchy
+use signed packages can be controlled by simply overriding the signing intent of
+the top most image. The signing intent of the layered images would then be
```

Comment:
People imagine layers different ways up depending on how their brains work. Is there a way of wording this that avoids "top"?

Reply:
I'm rattling my brain on this one :)

How about "top most ancestor image" ? It might better clarify which direction we mean.

Reply:
Yes, that's better.

Reply:
Done.